app/core/Tracker2/FullStoryTracker.js

```diff
@@ -0,0 +1,175 @@
+    o=n.createElement(t);o.async=1;o.crossOrigin='anonymous';o.src='https://'+_fs_script;
+    y=n.getElementsByTagName(t)[0];y.parentNode.insertBefore(o,y);
+    g.identify=function(i,v,s){g(l,{uid:i},s);if(v)g(l,v,s)};g.setUserVars=function(v,s){g(l,v,s)};g.event=function(i,v,s){g('event',{n:i,p:v},s)};
+    g.anonymize=function(){g.identify(!!0)};
+    g.shutdown=function(){g("rec",!1)};g.restart=function(){g("rec",!0)};
+    g.log = function(a,b){g("log",[a,b])};
+    g.consent=function(a){g("consent",!arguments.length||a)};
+    g.identifyAccount=function(i,v){o='account';v=v||{};v.acctId=i;g(o,v)};
+    g.clearUserCookie=function(){};
+    g._w={};y='XMLHttpRequest';g._w[y]=m[y];y='fetch';g._w[y]=m[y];
+    if(m[y])m[y]=function(){return g._w[y].apply(this,arguments)};
+    g._v="1.2.0";
+  })(window,document,window['_fs_namespace'],'script','user');
+}
+
+export default class FullstoryTracker extends BaseTracker {
+  constructor (store, globalTracker) {
+    super()
+
+    this.store = store
+    this.globalTracker = globalTracker
+
+    const sessionEnabled = window.sessionStorage.getItem(FULLSTORY_SESSION_TRACKING_ENALBED_KEY)
+    this.enableDecisionMade = (sessionEnabled !== null)
+    this.enabled = (sessionEnabled === 'true')
+
+    this.log('initialized enabled', this.enabled)
+    this.log('initialized enable decision made', this.enabled)
+  }
+
+  async _initializeTracker () {
+    this.watchForDisableAllTrackingChanges(this.store)
+
+    // TODO handle disable all tracking
+    window['_fs_ready'] = () => {
+      try {
+        this.log('ready')
+        if ((new URLSearchParams(window.location.search)).has(FULLSTORY_ENABLE_QUERY_PARAM)) {
+          this.enabled = true
+          this.log('query param force enable')
+        } else if (!this.enableDecisionMade) {
+          this.log('deciding on enabled')
+          this.enabled = this.decideEnabled()
+          if (this.enabled) {
+            this.globalTracker.trackEvent('FullStory Tracking Enabled')
+          }
+        }
+
+        if (this.enabled) {
+          this.log('enabling from init')
+          this.enable()
+        } else {
+          this.log('disabling from init')
+          this.disable()
+        }
+
+        this.onInitializeSuccess()
+      } catch (e) {
+        this.onInitializeFail(e)
+        throw e
+      }
+    }
+
+    loadFullStory()
+  }
+
+  enable () {
+    this.enabled = true
+    window.sessionStorage.setItem(FULLSTORY_SESSION_TRACKING_ENALBED_KEY, 'true')
+    FS.restart()
+    this.log('enabled')
+  }
+
+  disable () {
+    this.enabled = false
+    window.sessionStorage.setItem(FULLSTORY_SESSION_TRACKING_ENALBED_KEY, 'false')
+    FS.shutdown()
+    this.log('disabled')
+  }
+
+  decideEnabled () {
+    const { me } = this.store.state
+
+    if (this.disableAllTracking) {
+      this.log('decide enabled', 'disable all tracking')
+      return false
+    } else if (me.anonymous && Math.random() < 0.02) {
```

Comment:
We have about 500 - 2500 users active per minute. What sort of volume do we want to capture for fullstory?

Reply:
We're going to start by tracking:
 - 2% of teacher sessions (expecting ~ 1100 sessions for the month based on a rough estimate of 55k unique /teacher/classes pageviews last month - couldn't find a way to find number of teacher sessions)
- .25% of anonymous sessions (expecting 1800 sessions for the month based on the 730k sessions we had last month)